src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
```

Comment:
```suggestion
  for (const action of actions) {
```
I think it would be cleaner and easier for you.
You can also destructure the action to get all the needed props of it (type, extraData, keysToRemove)

Reply:
still not fixed. let's use `for of` construction